spec/NewTwitchApi/Resources/EventSubApiSpec.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace spec\NewTwitchApi\Resources;
+
+use GuzzleHttp\Client;
+use GuzzleHttp\Psr7\Request;
+use GuzzleHttp\Psr7\Response;
+use PhpSpec\ObjectBehavior;
+use Psr\Http\Message\ResponseInterface;
+
+class EventSubApiSpec extends ObjectBehavior
+{
+    private string $bearer = 'TEST_TOKEN';
+    private string $secret = 'SECRET';
+    private string $callback = 'https://example.com/';
+
+    private function generateRequest(string $type, string $version, array $condition): Request {
+        $bodyParams = [];
+
+        $bodyParams[] = ['key' => 'type', 'value' => $type];
+        $bodyParams[] = ['key' => 'version', 'value' => $version];
+        $bodyParams[] = ['key' => 'condition', 'value' => $condition];
+        $bodyParams[] = [
+            'key' => 'transport',
+            'value' => [
+                'method' => 'webhook',
+                'callback' => $this->callback,
+                'secret' => $this->secret,
+            ],
+        ];
+
+        return new Request('POST', 'eventsub/subscriptions', ['Authorization' => sprintf('Bearer %s', $this->bearer)], $bodyParams);
```

Comment:
Easy mistake, the body parameters are invalid in a Spec. You need to prepare them as if the API was sending it. Here is the correct code:
```php
$bodyParams = [];

        $bodyParams['type'] = $type;
        $bodyParams['version'] = $version;
        $bodyParams['condition'] = $condition;
        $bodyParams['transport'] = [
          'method' => 'webhook',
          'callback' => $this->callback,
          'secret' => $this->secret,
        ];

        return new Request('POST', 'eventsub/subscriptions', ['Authorization' => sprintf('Bearer %s', $this->bearer)], json_encode($bodyParams));
```